### PR TITLE
feat: integrate x/feemarket

### LINKED ISSUE #245: Add feemarket to eve
Please add it to eve
https://github.com/skip-mev/feemarket

## Todo
- [ ] Integrate to eve, add unit test(#250)
- [ ] Add integration test
- [ ] Add upgrade handler
- [ ] Add upgrade test

### PR DESCRIPTION
**Pull request by tropicaldog**
## Description

Partially addresses #245

## Summary by CodeRabbit

- **New Features**
  - Introduced a Fee Market module to manage transaction fees, including denomination resolution and fee calculation.
  
- **Bug Fixes**
  - Adjusted static check warnings for bank and staking parameter key tables.

- **Tests**
  - Added tests for fee handling logic with various scenarios and denominations.

- **Upgrades**
  - Implemented new upgrades to include the Fee Market module and associated configurations.

### CODE REVIEW

**Comment by faddat** [comment]
@coderabbitai review

**Comment by faddat** [comment]
In upgrades.go, there are a number of deprecated calls.  Are these needed?

I'd like to get this PR landed asap if possible.

File: app/upgrades/v1/constants.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"github.com/eve-network/eve/app/upgrades"
+	feemarkettypes "github.com/skip-mev/feemarket/x/feemarket/types"
+
+	store "cosmossdk.io/store/types"
+)
+
+const (
+	// UpgradeName defines the on-chain upgrade name.
+	UpgradeName = "v1"
```
Comment (hoank101) [inline app/upgrades/v1/constants.go:12]: looks good please change it to v0.0.6, also change folder name

**Comment by hoank101** [comment]
also we need add v0.0.6 upgrade to here -> https://github.com/eve-network/eve/blob/main/app/upgrades.go#L28

File: app/upgrades/v1/constants.go (hunk 1 above)
Comment (hoank101) [reply app/upgrades/v1/constants.go:12]: also we need add v0.0.6 upgrade to here -> https://github.com/eve-network/eve/blob/main/app/upgrades.go#L28

Comment (tropicaldog) [reply app/upgrades/v1/constants.go:12]: I think v0.1.0 is more suitable since we are adding a new module

Comment (hoank101) [reply app/upgrades/v1/constants.go:12]: let do it, thank you

**Review by faddat — APPROVED** [review]

**Pull request merged**